content/en/blog/2020/large-scale-security-policy-performance-tests/index.md

```diff
@@ -0,0 +1,148 @@
+
+## Test cases
+
+1. MTLS STRICT vs plaintext.
+
+1. A single authorization policy with a variable number of principal rules. For the RBAC filter to activate this rule a peer authentication policy must also be applied to the system.
+
+1. Authorization policy with a variable number of `requestPrincipal` rules. For the RBAC filter to activate this rule a `requestAuthentication` policy must also be applied to the system.
+
+1. A single authorization policy with variable number of paths vs `sourceIP` rules
+
+1. Variable number of authorization policies each consisting of a single paths or `sourceIP` rule
+
+1. A single `requestAuthentication` policy with variable number of `JWTRules` rules
+
+## Data
+
+The y axis of each test is the latency in milliseconds, and the x axis is the number of concurrent connections. The x axis of each graph will consist of 3 data points. This first being a small load (qps=100, conn=8), the second a medium load (qps=500, conn=32), and the third being a large load (qps=1000, conn=64)
+
+{{< tabset category-name="platform" >}}
+
+{{< tab name="MTLS vs plainText" category-value="one" >}}
+{{< image width="90%" ratio="45.34%"
+    link="./mtls_plaintext.svg"
+    alt="MTLS vs plaintext"
+    caption=""
+    >}}
+The difference between MTLS mode STRICT vs that of plaintext is very small in lower loads and as the number of qps and conn increase the latency of requests with MTLS STRICT rise higher. The additional latency increased in larger loads is minimal compared to that of the increase from having no sidecars vs having sidecars in the plaintext.
+{{< /tab >}}
+
+{{< tab name="AuthZ principals" category-value="two" >}}
+{{< image width="90%" ratio="45.34%"
+    link="./AuthZ_var_principals.svg"
+    alt="Authorization policy variable number of principals"
+    caption=""
+    >}}
+
+For Authorization policies with 10 vs 1000 principal rules, there is a larger increase of latency of having 10 principal rules compared to having no policies than there is with the increase of latency of having 1000 principals compared to having 10 principals.
+{{< /tab >}}
+
+{{< tab name="AuthZ requestPrincipals" category-value="three" >}}
+{{< image width="90%" ratio="45.34%"
+    link="./AuthZ_var_requestPrincipals.svg"
+    alt="Authorization policy with variable principals"
+    caption=""
+    >}}
+    For Authorization policies with a variable number of `requestPrincipal` rules, the difference of latency of having 10 `requestPrincipal` rules compared to no policies is nearly the same as the increase of having 1000 `requestPrincipal` rules compared to having 10 `requestPrincipal` rules.
+{{< /tab >}}
+
+{{< tab name="paths vs sourceIP" category-value="four" >}}
```

Comment:
Can we split this into two? one tab for just "AuthZ Path" and another tab for "AuthZ Source IP"

Reply:
Yes updated, but kept the image of the comparison of both in both tabs